<!doctype html>
<html lang="en">

<head>
	<meta charset="utf-8">
	<META HTTP-EQUIV="CACHE-CONTROL" CONTENT="NO-CACHE">

	<title>Interactive Data Visualization for the Web</title>

	<meta name="description" content="A short introduction to Data Vis with HTML5">
	<meta name="author" content="Dick Kreisberg">

	<meta name="apple-mobile-web-app-capable" content="yes" />
	<meta name="apple-mobile-web-app-status-bar-style" content="black-translucent" />

	<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

	<link href='http://fonts.googleapis.com/css?family=Lato:400,700,400italic,700italic' rel='f' type='text/css'>

	<link rel="stylesheet" href="css/reveal.css">
	<link rel="stylesheet" href="css/theme/night.css" id="theme">
	<link rel="stylesheet" href="css/custom.css" >

	<!-- For syntax highlighting -->
	<link rel="stylesheet" href="lib/css/zenburn.css">

	<script>
			// If the query includes 'print-pdf' we'll use the PDF print sheet
			document.write( '<link rel="stylesheet" href="css/print/' + ( window.location.search.match( /print-pdf/gi ) ? 'pdf' : 'paper' ) + '.css" type="text/css" media="print">' );</script>

	<!-- Font Awesome - http://fortawesome.github.com/Font-Awesome -->
	<link rel="stylesheet" href="lib/assets/css/font-awesome.css">

	<link href="lib/assets/css/vq.css" type="text/css" rel="stylesheet" />

	<script type="text/javascript" src="lib/js/d3/d3.js"></script>
	<script src="//ajax.googleapis.com/ajax/libs/jquery/1.7.2/jquery.min.js" type="text/javascript"></script>

	<!--[if IE 7]>
	<link rel="stylesheet" href="lib/assets/css/font-awesome-ie7.css">
	<![endif]-->

	<!--[if lt IE 9]>
	<script src="lib/js/html5shiv.js"></script>
	<![endif]-->
</head>

<body>

	<div class="reveal">

		<!-- Used to fade in a background when a specific slide state is reached -->
		<div class="state-background"></div>

		<!-- Any section element inside of this container is displayed as a slide -->
		<div class="slides">
			<section>
				<h2>Visualizing Multivariate Analysis of Cancer Data</h2>
				<p style="text-align:right;">
					<br/>
					Dick Kreisberg
					<br/>
					Institute for Systems Biology
					<br/>
					February 22, 2013
					<br/>
					<br/>
					<small>http://vis.systemsbiology.net/cancer_symposium/</small>
				</p>
			</section>
			<section>
				<a href="http://explorer.cancerregulome.org/all_pairs/index_local.html?dataset=brca_03feb13_seq_tumor_only&t_label=BRCA1&trans=true" target="_blank">
					In Breast Cancer, which interchromosomal molecular features are most associated with TP53 mutations?
				</a>
			</section>
			<aside class="notes">
				Chr 22, mir-hsa-3200, associated with ERBB2/Her2 gene in paired tissues
			</aside>
			<article title="Outline"></article>
			<section>
				<h2>Outline</h2>
				<p>
					<ul>
						<li>Introduction</li>
						<li>The Cancer Genome Atlas</li>
						<li>Multivariate Analysis of Heterogeneous Data</li>
						<li>Regulome Explorer</li>
						<li>Pubcrawl</li>
						<li>Into the Future</li>
					</ul>
				</p>

				<aside class="notes"></aside>
			</section>

			<section>
				<section>
					<h2>
						Why do we need tools that
						<q>explore</q>
						cancer data?
					</h2>
					<p>
						<ul>
							<li class="fragment">Lower the barrier to the data</li>
							<li class="fragment">
								Manage the three hurdles of data...
								<br> <em>Scope</em>
								, <em>Scale</em>
								,
								<em>Complexity</em>
							</li>
							<li class="fragment">Provide supporting data and context</li>
							<li class="fragment">Distribute the effort across the research community</li>
							<li class="fragment">Reduce the time to answer simple questions</li>
						</ul>
					</p>
					<aside class="notes">
					Barriers: Data is messy. Strange formats.  Different between tumors.
					</aside>
				</section>
				<section></section>
			</section>
			<article title="TCGA"></article>
			<section>
				<section>
					<h2>The Cancer Genome Atlas</h2>
					<img height=560 src="images/tcga_graph.png">
					<br/>
					<small>http://cancergenome.nih.gov/</small>
				</section>
				<section>
					<img src="images/tumor_barchart.png"></section>
				<section>
					<h3>TCGA Tumor Data</h3>
					<small>
						<div class="fragment" style="text-align:left;">
							Clinical information (age, gender, vital status, tumor grade, histology)
							<br/>
							Tumor sample high-throughput molecular data
							<br/>
						</div>
						<br/>
						<div class="fragment" style="text-align:left;">
							mRNA expression (Agilent arrays and/or Illumina GA/HiSeq)
							<br/>
							microRNA expression (Illumina GA/HiSeq)
							<br/>
							protein expression (Reverse Phase Protein Array)
							<br/>
						</div>
						<br/>
						<div class="fragment" style="text-align:left;">
							DNA methylation (Illumina 27k/450k)
							<br/>
							DNA copy-number segmentation (Affymetrix SNP6)
							<br/>
							DNA mutations (Illumina)
							<br/>
						</div>
						<br/>
						<div class="fragment" style="text-align:left;">
							<h4>Downstream Information:</h4>
							Subtype / Cluster assignments (single-platform analysis)
							<br/>
							GISTIC regions of interest (copy-number)
							<br/>
							Tumor Sample purity / ploidy
							<br/>
							Somatic Mutation rates
							<br/>
							Micro-Satellite Instability
							<br/>
						</div>
					</small>
				</section>
				<section>
					<h2>The Reality of Research Data:</h2>
					<ul>
						<li class="fragment roll-in">messy</li>
						<li class="fragment roll-in">noisy</li>
						<li class="fragment roll-in">contradictive</li>
						<li class="fragment roll-in">nuanced</li>
					</ul>
				</section>
			</section>
			<article title="Multivariate Analysis"></article>
			<section>
				<section>
					<h2>Multivariate Analysis</h2>
					<p>
						There are many methods being used in cancer research. In general, we see X types of approaches.
						<ol>
							<li class="fragment roll-in">Statistical</li>
							<li class="fragment roll-in">Matrix Decomposition</li>
							<li class="fragment roll-in">Information Theoretic</li>
							<li class="fragment roll-in">Machine Learning</li>
						</ol>
					</p>
				</section>
				<section>
					<h3>Heterogeneous Feature Matrix</h3>
					<img height=600 src="images/feature_matrix.png"></section>
				<section>
					<h2>Association Analysis</h2>
					<img height = 600 src="images/feature_analysis.png"></section>
				<section>
					<h2>Pairwise Analysis</h2>
					<img src="images/feature_feature.png" height=300>
					<br>
					<ul>
						<li class="fragment">
							Continuous - Continuous: F Statistic
							<br/>
							<small>RNAseq of BRCA2 -></small>
						</li>
						<li class="fragment">
							Continuous - Categorical: ANOVA
							<br/>
							<small>miRNA hsa-mir-xxx -></small>
						</li>
						<li class="fragment">
							Categorical - Categorical: Fisher's Exact Test
							<br/>
							<small>PAM50_call4 -></small>
						</li>
						<li class="fragment">Bonferroni correction applied</li>
					</ul>
				</section>
				<section>
					<h2>Pairwise Edges</h2>
				</section>
			</section>
			<section>
				<section>
					<h2>Random Forest</h2>
					Decision Tree Ensemble Learning
					<br>
					<br/><br/>
					RF-ACE: Random Forest with Artificial Contrast Ensembles<br/>
					<small>developed by Timo Erkkila</small><br/>
					<small>http://rf-ace.googlecode.com</small>
					<br/><br/>
					<br/><br/>
					<small>
						Breiman, Leo (2001). "Random Forests", <i>Machine Learning</i>
						, <b>45</b>
						(1):5-32.
					</small>
						<small>Tuv, Eugene, et al. "Feature selection with ensembles, artificial variables, and redundancy elimination." <i>The Journal of Machine Learning Research</i> <b>10</b> (2009): 1341-1366.</small>
				</section>
				<section>
					<h2>A Decision Tree</h2>
					<img width=1200 src="images/decision_tree.png"></section>
				<section>
					<h2>Randomness - Bootstrapping</h2>
					<img src="images/bootstrap.png">
					<br/>
				</section>
				<section>
					<h2>Randomness - Bagging</h2>
					<img src="images/bagging.png"></section>
				<section>
					<h2>A Forest of Voting Trees</h2>
					<img height=600 src="images/random_forest.png"></section>
			</section>
			<section>
				<section>
					<h3>Multi-scale Association Explorer</h3>
					<iframe frameborder="0" height="800" width="960" src="http://explorer.cancerregulome.org/all_pairs/index.html?dataset=brca_manuscript_rerun_nov12d_pw&t_type=CNVR&p_type=GEXP&trans=true"></iframe>
				</section>
				<section>
					<h3>Edges Between Genomic Features</h3>
					<div id="circvis"></div>
				</section>
				<section>
					<h3>Managing Scale</h3>
				</section>
				<section>
					<h3>Edges With Non-Genomic Features</h3>
				</section>
				<section>
					<h3>Analytical Integrity</h3>
					(scatterplot, violinplot, cubbyhole)
				</section>
			</section>
			<section>
				<section>
					<h2>Colorectal Cancer Aggressiveness</h2>
					A combined p-value approach to identifying molecular features associated with tumor aggresiveness.<br/>
					<small>Vesteinn Thorsson</small>
				</section>
			</section>
			<section>
				<section>
					<h2>PubCrawl</h2>
					Incorporating Semantic and Interaction Associations<br/>
					<small>Andrea Eakin, Brady Bernard</small>
				</section>
				<section>Normalized Google Distance</section>
				<section>Protein Domain Interactions</section>
			</section>
			<section>
				<section>
					<h2>Into the Future</h2>
					There are new dimensions of the data and computation to explore.
				</section>
				<section>
					<h3>Tumor Type (and Subtype)</h3>
				</section>
				<section>
					<h3>Unsupervised Clustering of Samples</h3>
					Naive creation of Cohorts
				</section>
				<section>
					<h3>High-Throughput Computation</h3>
					<img width=800 src="images/google_io1.jpg">
					<br/>
					600,000 cores running Random Forest on Google Compute Engine
				</section>
			</section>
			<section>
				<section></section>
				<section></section>
			</section>
			<section>
				<section>
					<h2>Resources</h2>
					<table>
						<tbody>
							<tr>
								<td>d3.js</td>
								<td>d3js.org</td>
							</tr>
							<tr>
								<td>Science.js</td>
								<td>github.com/jasondavies/science.js/</td>
							</tr>
							<tr>
								<td>CytoscapeWeb</td>
								<td>http://cytoscapeweb.cytoscape.org/</td>
							</tr>
							<tr>
								<td>Cytoscape.js</td>
								<td>http://cytoscape.github.com/cytoscape.js/</td>
							</tr>
						</tbody>
					</table>
				</section>
				<section>
					<h3>The Center for Systems Analysis of the Cancer Regulome</h3>
					Ilya Shmulevich (ISB) + Wei Zhang (MD Anderson Cancer Center)
					<div style="height:500px">
						<img style="position:absolute;
								top:160px;
								left:0px;"
								width=300
								src="images/cancer_cell_cover.jpg">
						<img style="position:absolute;
								top:200px;
								left:290px;
								-ms-transform:rotate(1deg);
								-moz-transform:rotate(1deg);
								-webkit-transform:rotate(1deg);
								-o-transform:rotate(1deg);"
								width=600
								src="images/breast_cancer_article.png">
						<img style="position:absolute;
								top:340px;
								left:295px;
								-ms-transform:rotate(-2deg);
								-moz-transform:rotate(-2deg);
								-webkit-transform:rotate(-2deg);
								-o-transform:rotate(-2deg);"
								width=610
								src="images/coad_cancer_article.png">
						<img style="position:absolute;
								top:330px;
								left:540px;"
								height=40
								src="images/nature_logo.png">
						<img style="position:absolute;
								top:180px;
								left:390px;
								-ms-transform:rotate(4deg);
								-moz-transform:rotate(4deg);
								-webkit-transform:rotate(4deg);
								-o-transform:rotate(4deg);" 
								width=650
								src="images/fastbreak_article.png">
						<img style="position:absolute;
								top:500px;
								left:440px;"
								height=160
								src="images/jama_article.png">
						<img style="position:absolute;
								top:560px;
								left:30px;"
								width=400
								src="images/bernard_article.png"></div>
					<br/>
					<small>www.cancerregulome.org</small>
				</section>
				<section>
					<h2>Projects @ ISB</h2>
					<p>
						Regulome Explorer
						<br/>
						<a href="http://explorer.cancerregulome.org" target="_blank">explorer.cancerregulome.org</a>
						<br/>
						Pubcrawl
						<br/>
						<a href="pubcrawl.systemsbiology.net" target="_blank">pubcrawl.systemsbiology.net</a>
						<br/>
						Genespot
						<br/>
						<a href="www.genespot.net" target="_blank">www.genespot.org</a>
						<br/>
						Transcriptional Regulation & Epigenetic Landscape
						<br/>
						<a href="http://trel.systemsbiology.net" target="_blank">trel.systemsbiology.net</a>
						<br/>
						<br/>
					</p>
					<br/>
					Search 'codefor@systemsbiology.org' at code.google.com
				</section>
				<section>
					<h3>Acknowledgements</h3>
					<table style="width:100%;font-size:22px;line-height:23px;">
						<col style="width:33%"/>
						<col style="width:33%"/>
						<col style="width:33%"/>
						<tbody>
							<tr>
								<td>
									<img height=90 src="images/isb_logo.png"></td>
								<td>&nbsp;</td>
								<td>
									<img height=90 src="images/mdanderson_logo.png"></td>
							</tr>
							<tr>
								<td> <b>Ilya Shmulevich</b>
								</td>
								<td>&nbsp;</td>
								<td>Wei Zhang</td>
							</tr>
							<tr>
								<td>&nbsp;</td>
								<td>&nbsp;</td>
								<td>&nbsp;</td>
							</tr>
							<tr>
								<td>Andrea Eakin</td>
								<td>Hector Rovira</td>
								<td>Da Yang</td>
							</tr>
							<tr>
								<td>Jake Lin</td>
								<td>Ryan Bressler</td>
								<td>Yuexin Lin</td>
							</tr>
							<tr>
								<td>Brady Bernard</td>
								<td>Timo Erkkila</td>
								<td>Yan Sun</td>

							</tr>
							<tr>
								<td>Sheila Reynolds</td>
								<td>Vesteinn Thorsson</td>
								<td>&nbsp;</td>
							</tr>
							<tr>
								<td>Lisa Iype</td>
								<td>Kalle Leinonen</td>
								<td>&nbsp;</td>
							</tr>
							<tr>
								<td>Patrick May</td>
								<td>Lesley Wilkerson</td>
							</tr>
						</tbody>
					</table>
					<br/>
					<table>
						<tbody>
							<tr>
								<td>
									<img height=60 style="background:#FFFFFF" src="images/tcga_logo.png"></td>
								<td>
									<img height=90 src="images/nci_logo.png"></td>
								<td>
									<img height=60 src="images/nhgri_logo.png"></td>
							</tr>
						</tbody>
					</table>

				</section>
				<section>
					<br/>
					<br/>
					<br/>
					<code>dick.kreisberg@systemsbiology.org</code>
				</section>
			</section>
			<section>
				<section>
					<h2>Questions?</h2>
				</section>
				<section>
					<p>
						<h3>Reveal.JS</h3>
						lab.hakim.se/reveal-js/
					</p>
					<p>
						<h3>JSFiddle</h3>
						jsfiddle.net
					</p>
					<p>
						<h3>NodeJS</h3>
						nodejs.org
					</p>
					<p>
						<h3>Circos</h3>
						circos.ca
					</p>
				</section>
			</section>
		</div>
		<!-- Presentation progress bar -->
		<div class="progress">
			<span></span>
		</div>
	</div>

	<script src="lib/js/head.min.js"></script>
	<script src="js/reveal.js"></script>

	<script>		
	// Full list of configuration options available here:
	// https://github.com/hakimel/reveal.js#configuration
	Reveal.initialize({
		controls: true,
		progress: true,
		history: true,
		center: true,
		
		theme: Reveal.getQueryHash().theme || 'night', // available themes are in /css/theme
		transition: Reveal.getQueryHash().transition || 'linear', // default/cube/page/concave/linear(2d)

		// Optional libraries used to extend on reveal.js
		dependencies: [
			{ src: 'lib/js/highlight.js', async: true, callback: function() { window.hljs.initHighlightingOnLoad(); } },
			{ src: 'lib/js/classList.js', condition: function() { return !document.body.classList; } },
			{ src: 'plugin/markdown/showdown.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
			{ src: 'plugin/markdown/markdown.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
			{ src: 'plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } },
			{ src: 'plugin/zoom-js/zoom.js', async: true, condition: function() { return !!document.body.classList; } },
				{ src: 'plugin/notes/notes.js', async: true, condition: function() { return !!document.body.classList; } }
			// { src: 'plugin/remotes/remotes.js', async: true, condition: function() { return !!document.body.classList; 
		]
	});
	
</script>
	<script type="text/javascript" src="lib/js/underscore/underscore.js"></script>
	<script type="text/javascript" src="lib/js/vq/vq.js"></script>
	<script type="text/javascript" src="lib/js/vq/vq.circvis.js"></script>
	<script type="text/javascript" src="lib/circvis/data/circvis_data.json"></script>
	<script type="text/javascript" src="lib/circvis/js/circvis_demo.js"></script>
	<script type="text/javascript" src="lib/circvis/js/circvis_configure.js"></script>
	<script type="text/javascript">
      var circle;
    // circle = circvis.plot($('#wedge1').get(0));
</script>

</body>
</html>